config.go

```diff
@@ -869,6 +873,9 @@ func (cfg *Config) renewCert(ctx context.Context, name string, force, interactiv
 				}
 			}
 
+			// the ACME client will use this to tell the server we are replacing a certificate
+			ctx = context.WithValue(ctx, ctxKeyARIReplaces, leaf)
```

Comment:
Could this lead to a refusal if the `leaf` was signed by another issuer?
For instance:
- initial certificate signed by ZeroSSL
- upon renewal ZeroSSL is down, so fallback to letsencrypt
- letsencrypt didn't know about this `leaf` certificate, so if I understand https://letsencrypt.org/2024/04/25/guide-to-integrating-ari-into-existing-acme-clients#step-6-indicating-which-certificate-is-replaced-by-this-new-order correcty, it will reject the request

> When Let’s Encrypt processes a new order request featuring a ‘replaces’ field, several important checks are conducted. First, it’s verified that the certificate indicated in this field has not been replaced previously. Next, we ensure that the certificate is linked to the same ACME account that’s making the current request. Additionally, there must be at least one domain name shared between the existing certificate and the one being requested. If these criteria are met and the new order request is submitted within the ARI-suggested renewal window, the request qualifies for exemption from all rate limits. 

This is also suggested by the standard: https://datatracker.ietf.org/doc/draft-ietf-acme-ari/

> Servers SHOULD check that the identified certificate and the current
> New Order request correspond to the same ACME Account and share a
> preponderance of identifiers, and that the identified certificate has
> not already been marked as replaced by a different finalized Order.
> Servers MAY ignore the replaces field in New Order requests which do
> not pass such checks.

(they can't belong to the same ACME account if they are from different issuers)

Reply:
Funny puppy being this up, I already have a patch that Flores this locally. I just need to test it and commit it.

Reply:
Thanks for the analysis btw, great writeup/catch!